diff --git a/KrkrPatch/dllmain.cpp b/KrkrPatch/dllmain.cpp
--- a/KrkrPatch/dllmain.cpp
+++ b/KrkrPatch/dllmain.cpp
@@ -279,13 +279,19 @@ void OnShutdown()
 
 
 BOOL APIENTRY DllMain(HMODULE hModule, DWORD  ul_reason_for_call, LPVOID lpReserved)
 {
+	if (DetourIsHelperProcess())
+	{
+		return TRUE;
+	}
+
 	switch (ul_reason_for_call)
 	{
 		case DLL_PROCESS_ATTACH:
 		{
+			DetourRestoreAfterWith();
 			g_hEXE = GetModuleHandle(NULL);
 			g_hDLL = hModule;
 			OnStartup();
 			break;
 		}
diff --git a/KrkrPatchLoader/KrkrPatchLoader.cpp b/KrkrPatchLoader/KrkrPatchLoader.cpp
index 0ef4e8e..7b97c42 100644
--- a/KrkrPatchLoader/KrkrPatchLoader.cpp
+++ b/KrkrPatchLoader/KrkrPatchLoader.cpp
@@ -20,0 +21 @@ int WINAPI WinMain(_In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance, _
+	std::wstring steamAppId;
@@ -37,0 +39 @@ int WINAPI WinMain(_In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance, _
+		auto steamAppId_ = Encoding::Utf8ToUtf16(cfg.value("steamAppId", std::string()));
@@ -46,0 +49 @@ int WINAPI WinMain(_In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance, _
+		steamAppId = steamAppId_;
@@ -61,0 +65,6 @@ int WINAPI WinMain(_In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance, _
+
+	if (!steamAppId.empty())
+	{
+		SetEnvironmentVariableW(L"SteamAppId", steamAppId.c_str());
+		SetEnvironmentVariableW(L"SteamGameId", steamAppId.c_str());
+	}
